library/nmcli.py

```diff
@@ -0,0 +1,735 @@
+            type=dict(required=False, default=None, choices=['ethernet', 'team', 'team-slave', 'bond', 'bond-slave', 'bridge', 'vlan'], type='str'),
+            ip4=dict(required=False, default=None, type='str'),
+            gw4=dict(required=False, default=None, type='str'),
+            dns4=dict(required=False, default=None, type='str'),
+            ip6=dict(required=False, default=None, type='str'),
+            gw6=dict(required=False, default=None, type='str'),
+            dns6=dict(required=False, default=None, type='str'),
+            # general usage
+            mtu=dict(required=False, default=None, type='str'),
+            mac=dict(required=False, default=None, type='str'),
+            # bridge specific vars
+            stp=dict(required=False, default='yes', choices=['yes', 'no'], type='str'),
+            priority=dict(required=False, default="128", type='str'),
+            slavepriority=dict(required=False, default="32", type='str'),
+            forwarddelay=dict(required=False, default="15", type='str'),
+            hellotime=dict(required=False, default="2", type='str'),
+            maxage=dict(required=False, default="20", type='str'),
+            ageingtime=dict(required=False, default="300", type='str'),
+            # vlan specific vars
+            vlanid=dict(required=False, default=None, type='str'),
+            # following options are specific to ifnamedel
+        ),
+        supports_check_mode=True
+    )
+
+    # if module.params['state'] is None and module.params['enabled'] is None:
+    #     module.fail_json(msg="'Neither state' nor 'enabled' is set")
+
+    nmcli=Nmcli(module)
+
+    if nmcli.syslogging:
+        syslog.openlog('ansible-%s' % os.path.basename(__file__))
+        syslog.syslog(syslog.LOG_NOTICE, 'Nmcli instantiated - platform %s' % nmcli.platform)
+        if nmcli.distribution:
+            syslog.syslog(syslog.LOG_NOTICE, 'Nuser instantiated - distribution %s' % nmcli.distribution)
+
+    rc=None
+    out=''
+    err=''
+    result={}
+    result['cname']=nmcli.cname
+    result['state']=nmcli.state
+
+    if nmcli.state=='absent':
+        # if nmcli.connection_exists():
+        if module.check_mode:
+            module.exit_json(changed=True)
+        (rc, out, err)=nmcli.connection_down()
+        (rc, out, err)=nmcli.remove_connection()
+
+        # elif not nmcli.connection_exists():
+        #     result['Connection']='No Connection named %s exists' % nmcli.cname
+        #     (rc, out, err)=nmcli.connection_down()
+        #     (rc, out, err)=nmcli.remove_connection()
+        #     result['Connection']='Connection %s is being removed' % nmcli.cname
+        #     result['Connection']='Yessss. Connection %s exists' % nmcli.cname
+        #     result['Connection']='Connection %s is being set to down' % nmcli.cname
+        if rc!=0:
+            module.fail_json(name =('No Connection named %s exists' % nmcli.cname), msg=err, rc=rc)
```

Comment:
Don't fail when a connection that is specified as absent is already absent. exit with changed=False.